socialhome/streams/app/components/StreamElement.vue

```diff
@@ -1,13 +1,17 @@
 <template>
     <div>
-        <div v-html="htmlSafe"></div>
+        <div v-if="nsfw" class="card card-block text-center nsfw-shield">
+            <b-button href="#" variant="link" @click.stop.prevent="toggleNsfw">
+                {{ toggleNsfwBtnText }}
+            </b-button>
+        </div>
+        <div v-if="showNsfw" v-html="htmlSafe"></div>
```

Comment:
This is incorrect. The NSFW shield is already in the rendered contents in Socialhome. There is an `nsfw` class added to all images in the content if a tag is found. This activates the NSFW shield. What you need in the JS side is a trigger to show it on click. There should also be a trigger then to return the nsfw shield. Try it to see how it works.

We don't want to hide the whole content, just images. Diaspora NSFW shield always annoyed me because you can't see any context to the post before clicking it.

Reply:
Hmm... I don't like that. It's not the Vue way. It will require messing with selectors :/

Reply:
Can't bend features to code.. Code must bend to features ;)